src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -323,16 +318,61 @@ class RideHailManager(
       val persons: Iterable[Person] = RandomUtils
         .shuffle(scenario.getPopulation.getPersons.values().asScala, rand)
         .take(numRideHailAgents.toInt)
+      val vehicleTypes = VehiclesAdjustment
+        .getVehicleAdjustment(beamServices)
+        .sampleRideHailVehicleTypes(
+          numVehicles = numRideHailAgents.toInt,
+          vehicleCategory = VehicleCategory.Car
+        )
+      var activityEndTimes: ArrayBuffer[Int] = new ArrayBuffer[Int]()
```

Comment:
Please do not use a var. It is the worst possible scenario for creating garbage. Use an immutable reference to a mutable buffer instead.